drivers: wifi: Notify app of local TWT tear down

ajayparida opened the pull request:
UMAC is not sending any feedback for the local teardown CMD, to keep the application design consistent, driver should always inform with the teardown status, it would be always success till UMAC implements the feedback.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13575/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13575/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-chip | X | | |
| test-sdk-wifi | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13575/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

sachinthegreen commented on drivers/wifi/nrf700x/src/wifi_mgmt.c:
```diff
@@ -428,6 +429,10 @@ int nrf_wifi_twt_teardown_flows(struct nrf_wifi_vif_ctx_zep *vif_ctx_zep,
 		}
 		/* UMAC doesn't send TWT teardown event for host initiated teardown */
 		nrf_wifi_twt_update_internal_state(vif_ctx_zep, false, flow_id);
+		/* Raise a tear down event from host itself, till UMAC supports */
```
Don't we need to encode the status i.e. `SUCCESS` or `FAIL` ?

ajayparida replied:
For host genearted, it will be always success. There is no command status field. Only for peer response of sucess/failure. 
Do I need to add for a command status field as this is temporary till UMAC gives proper event?

ajayparida replied:
Yeah, I think it's better to keep success/failure. I will update accordingly

krish2718 replied:
```suggestion
		/* TODO: Remove this once UMAC sends the status */
```

krish2718 replied:
Yes, we should always inform the status

ajayparida replied:
Yeah, I will update.

ioannisg commented:
@ajayparida @sachinthegreen is this event-notification to be demonstrated in the TWT sample (or elsewhere)?

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13575
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

ajayparida commented:
> @ajayparida @sachinthegreen is this event-notification to be demonstrated in the TWT sample (or elsewhere)?



> @ajayparida @sachinthegreen is this event-notification to be demonstrated in the TWT sample (or elsewhere)?

Yes in the sample twt app user will be  notified of the teardown status(success/failure) with flow id. 
Also in the shell application, once user initiates teardowm, will be notified.

rlubos commented:
@ajayparida We've been into lot of struggle due to broken BLE CI. I've updated manifest to the latest sdk-zephyr revision in https://github.com/nrfconnect/sdk-nrf/pull/13708 - please drop the `west.yml` change and we can focus on Wi-Fi specific part only, hopefully w/o fighting with CI anymore.

ajayparida commented:
> @ajayparida We've been into lot of struggle due to broken BLE CI. I've updated manifest to the latest sdk-zephyr revision in #13708 - please drop the `west.yml` change and we can focus on Wi-Fi specific part only, hopefully w/o fighting with CI anymore.

Dropped west.yml change.
